### PR TITLE
refac: Adjustments to design of views

### PR DESCRIPTION
**Pull request by dstenroejl**
## Description

- [x] Have a test data description class per CSV file with test data
- [x] Have a "view query" class per view

## References

Part of: https://app.zenhub.com/workspaces/mosaic-60a6105157304f00119be86e/issues/gh/energinet-datahub/team-mosaic/199

### CODE REVIEW

File: source/IntegrationTests/Application/OutgoingMessages/TestData/EnergyResultPerBrpGridAreaDescription.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    public EnergyResultPerBrpGridAreaDescription()
+        : base("balance_fixing_01-11-2022_01-12-2022_ga_543_per_brp_ga_v1.csv")
+    {
+    }
+
+    public override Guid CalculationId => Guid.Parse("9d4ff72b-fca9-44cd-a829-5b1d9766e1a0");
+
+    public override string GridAreaCode => "543";
+
+    public override int ExpectedOutgoingMessagesCount => 20;
```
Comment (lasrinnil) [inline source/IntegrationTests/Application/OutgoingMessages/TestData/EnergyResultPerBrpGridAreaDescription.cs:34]: Who has verified this is correct?

File: source/OutgoingMessages.Application/OutgoingMessagesClient.cs [hunk 2]
```diff
@@ -165,14 +166,50 @@ public async Task<int> EnqueueEnergyResultsForGridAreaOwnersAsync(EnqueueMessage
...
+    public async Task<int> EnqueueEnergyResultsForBalanceResponsiblesAndEnergySuppliersAsync(EnqueueMessagesInputDto input)
+    {
+        var numberOfEnqueuedMessages = 0;
+
+        var query = new EnergyResultPerEnergySupplierBrpGridAreaQuery(_energyResultEnumerator.EdiDatabricksOptions, input.CalculationId);
+        await foreach (var energyResult in _energyResultEnumerator.GetAsync(query))
+        {
+            // TODO: It should be possible to create the EnergyResultMessageDto directly in queries
+            var receiverNumber = await _masterDataClient.GetGridOwnerForGridAreaCodeAsync(energyResult.GridAreaCode, CancellationToken.None).ConfigureAwait(false);
+            var energyResultMessage = EnergyResultMessageDtoFactory.Create(EventId.From(input.EventId), energyResult);
```
Comment (lasrinnil) [inline source/OutgoingMessages.Application/OutgoingMessagesClient.cs:208]: We have to deliever a message to both parties pr. result.

first message is for the EnergySupplier
receiver -> EnergySuplierId
- message should NOT set the BalaneceResposibleId
second message is for the BalanceResponsibleParty

receiver -> BalanceResponsibleParty
- message should set EnergySupplierId

File: source/IntegrationTests/Application/OutgoingMessages/TestData/EnergyResultPerBrpGridAreaDescription.cs (hunk 1 above)
Comment (dstenroejl) [reply source/IntegrationTests/Application/OutgoingMessages/TestData/EnergyResultPerBrpGridAreaDescription.cs:34]: Mandalorian

File: source/OutgoingMessages.Application/OutgoingMessagesClient.cs (hunk 2 above)
Comment (dstenroejl) [reply source/OutgoingMessages.Application/OutgoingMessagesClient.cs:208]: Great, thx, I will fix it.

File: source/OutgoingMessages.Infrastructure/Databricks/EnergyResults/Factories/EnergyResultMessageDtoFactory.cs [hunk 3]
```diff
@@ -50,8 +50,63 @@ public class EnergyResultMessageDtoFactory()
...
+            eventId,
+            receiverNumber: receiverNumber,
+            receiverRole: receiverRole,
+            gridAreaCode: energyResult.GridAreaCode,
+            meteringPointType: energyResult.MeteringPointType.Name,
+            settlementMethod: energyResult.SettlementMethod?.Name,
+            measureUnitType: DomainModel.MeasurementUnit.Kwh.Name, // TODO: Should this be read from Databricks?
+            resolution: energyResult.Resolution.Name,
+            energySupplierNumber: null,
+            balanceResponsibleNumber: null,
```
[the remark this answers is not in the document]
Comment (dstenroejl) [reply source/OutgoingMessages.Infrastructure/Databricks/EnergyResults/Factories/EnergyResultMessageDtoFactory.cs]: Fixed

File: source/OutgoingMessages.Infrastructure/Databricks/EnergyResults/Factories/EnergyResultMessageDtoFactory.cs [hunk 4]
```diff
@@ -50,8 +50,63 @@ public class EnergyResultMessageDtoFactory()
...
```
[the remark this answers is not in the document]
Comment (dstenroejl) [reply source/OutgoingMessages.Infrastructure/Databricks/EnergyResults/Factories/EnergyResultMessageDtoFactory.cs:96]: Fixed

File: source/OutgoingMessages.Infrastructure/Databricks/EnergyResults/Factories/EnergyResultMessageDtoFactory.cs [hunk 5]
```diff
@@ -50,8 +49,85 @@ public class EnergyResultMessageDtoFactory()
...
+            energySupplierNumber: energyResult.EnergySupplierId,
+            balanceResponsibleNumber: null,
+            period: new DomainModel.Period(energyResult.PeriodStartUtc, energyResult.PeriodEndUtc),
+            points: CreateEnergyResultMessagePoints(energyResult.TimeSeriesPoints),
+            businessReasonName: businessReason.Name,
+            calculationResultVersion: energyResult.CalculationVersion,
+            settlementVersion: settlementVersion?.Name);
+    }
+
+    public static EnergyResultMessageDto CreateForBalanceResponsible(
```
Comment (lasrinnil) [inline source/OutgoingMessages.Infrastructure/Databricks/EnergyResults/Factories/EnergyResultMessageDtoFactory.cs:104]:
```suggestion
    public static EnergyResultMessageDto CreateForBalanceResponsiblePrEnergySupplier(
```
```suggestion
    public static EnergyResultMessageDto CreateForBalanceResponsible(
```

File: source/OutgoingMessages.Infrastructure/Databricks/EnergyResults/Queries/EnergyResultPerBrpGridAreaQuery.cs [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+using Energinet.DataHub.EDI.OutgoingMessages.Infrastructure.Databricks.DeltaTableConstants;
+using Energinet.DataHub.EDI.OutgoingMessages.Infrastructure.Databricks.EnergyResults.Factories;
+using Energinet.DataHub.EDI.OutgoingMessages.Infrastructure.Databricks.EnergyResults.Models;
+using Energinet.DataHub.EDI.OutgoingMessages.Infrastructure.Databricks.SqlStatements;
+using Energinet.DataHub.EDI.OutgoingMessages.Infrastructure.Extensions.Options;
+
+namespace Energinet.DataHub.EDI.OutgoingMessages.Infrastructure.Databricks.EnergyResults.Queries;
+
+/// <summary>
+/// The query we must perform on the 'nergy Result Per Balance Responsible Party, Grid Area' view.
```
Comment (lasrinnil) [inline source/OutgoingMessages.Infrastructure/Databricks/EnergyResults/Queries/EnergyResultPerBrpGridAreaQuery.cs:24]:
```suggestion
/// The query we must perform on the 'Energy Result Per Balance Responsible Party, Grid Area' view.
```
```suggestion
/// The query we must perform on the 'nergy Result Per Balance Responsible Party, Grid Area' view.
```

**Review by lasrinnil — APPROVED** [review]

**Pull request merged**